llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -307,6 +308,18 @@ void processFunction(Function &F, FunctionToAspectsMapTy &FunctionToAspects,
         CG[&F].insert(CI->getCalledFunction());
     }
   }
+
+  if (F.hasMetadata("sycl_used_aspects")) {
+    const MDNode *MD = F.getMetadata("sycl_used_aspects");
+    AspectsSetTy Aspects;
+    for (size_t I = 0, E = MD->getNumOperands(); I < E; ++I) {
+      Constant *C =
+        cast<ConstantAsMetadata>(MD->getOperand(I).get())->getValue();
+      auto Aspect = static_cast<uint32_t>(C->getUniqueInteger().getZExtValue());
```

Comment:
```suggestion
      auto Aspect = static_cast<uint32_t>(cast<ConstantInt>(C)->getSExtValue());
```

Let's use the same approach which is already used in the file. Please note that I also suggest `zext` -> `sext` change. It would be great to double-check with headers, SYCL spec and perhaps the design doc to see which one is correct. Note: `AspectsSetTy` stores `int`s, not `uint32_t`

Reply:
Done [a6da336](https://github.com/intel/llvm/pull/7035/commits/a6da336c7d130956827dc6a7c3c9a6fafd966aad)